keps/sig-api-machinery/4358-custom-resource-field-selectors/README.md

```diff
@@ -0,0 +1,1150 @@
+This is different from e2e tests which do not allow configuration of parameters.
+Doing this allows testing non-default options and multiple different and potentially conflicting command line options.
+-->
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+- test/integration/controlplane/crd_test.go
+  - TODO: Test selectable field {with, without valueIsAbsent}, {with single field, with multiple fields, with combined use of label selectors} for {list, watch}
+
+- <test>: <link to test coverage>
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+- test/e2e/apimachinery/custom_resource_definition.go
+  - TODO: Test selectable fields with combined multiple fields and combined with label selectors for {list, watch}
+
+- <test>: <link to test coverage>
+
+### Graduation Criteria
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Define graduation milestones.
+
+These may be defined in terms of API maturity, [feature gate] graduations, or as
+something else. The KEP should keep this high-level with a focus on what
+signals will be looked at to determine graduation.
+
+Consider the following in developing the graduation criteria for this enhancement:
+- [Maturity levels (`alpha`, `beta`, `stable`)][maturity-levels]
+- [Feature gate][feature gate] lifecycle
+- [Deprecation policy][deprecation-policy]
+
+Clearly define what graduation means by either linking to the [API doc
+definition](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-versioning)
+or by redefining what graduation means.
+
+In general we try to use the same stages (alpha, beta, GA), regardless of how the
+functionality is accessed.
+
+[feature gate]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+[deprecation-policy]: https://kubernetes.io/docs/reference/using-api/deprecation-policy/
+
+Below are some examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+--->
+
+#### Alpha
+
+- CRD selectableFields implemented
+- Feature implemented behind a feature flag
+- e2e tests completed and enabled
+
+#### Beta
+
+- built-in selectableFields converted to declarative go tags
+- built-in selectableFields included in OpenAPIv3
```

Comment:
this probably still makes sense if you can think of how to do it.

Reply:
I've moved built-in declarative support out of this KEP.  If I can find a way to support OpenAPIv3 with the built-ins as we have them, I'll try to get it in.  But for now, I'm scoping this KEP to CRD support, since that's probably all I can realistically get done.